managed/CustomGroup__event_messages_settings.php

```diff
@@ -0,0 +1,239 @@
+    'entity' => 'CustomField',
+    'cleanup' => 'always',
+    'update' => 'unmodified',
+    'params' => [
+      'version' => 4,
+      'values' => [
+        'custom_group_id.name' => 'event_messages_settings',
+        'name' => 'event_messages_custom_data_workaround',
+        'label' => E::ts('Custom Data Workaround'),
+        'data_type' => 'Boolean',
+        'html_type' => 'Radio',
+        'is_searchable' => TRUE,
+        'is_view' => TRUE,
+        'column_name' => 'custom_data_workaround',
+        'in_selector' => TRUE,
+      ],
+      'match' => [
+        'custom_group_id',
+        'name',
+      ],
+    ],
+  ],
+  [
+    'name' => 'CustomGroup_event_messages_settings_CustomField_event_messages_cc',
+    'entity' => 'CustomField',
+    'cleanup' => 'always',
+    'update' => 'unmodified',
+    'params' => [
+      'version' => 4,
+      'values' => [
+        'custom_group_id.name' => 'event_messages_settings',
+        'name' => 'event_messages_cc',
+        'label' => E::ts('CC'),
+        'html_type' => 'Text',
+        'is_searchable' => TRUE,
+        'column_name' => 'cc',
+        'in_selector' => TRUE,
+      ],
+      'match' => [
+        'custom_group_id',
+        'name',
+      ],
+    ],
+  ],
+  [
+    'name' => 'CustomGroup_event_messages_settings_CustomField_event_messages_reply_to',
+    'entity' => 'CustomField',
+    'cleanup' => 'always',
+    'update' => 'unmodified',
+    'params' => [
+      'version' => 4,
+      'values' => [
+        'custom_group_id.name' => 'event_messages_settings',
+        'name' => 'event_messages_reply_to',
+        'label' => E::ts('Reply-To'),
+        'html_type' => 'Text',
+        'is_searchable' => TRUE,
+        'column_name' => 'reply_to',
+        'in_selector' => TRUE,
+      ],
+      'match' => [
+        'custom_group_id',
+        'name',
+      ],
+    ],
+  ],
+  // TODO: Remove check when minimum core version requirement is >= 6.0.0.
+  (class_exists('\Civi\Api4\SiteEmailAddress')
+    ? [
+      'name' => 'CustomGroup_event_messages_settings_CustomField_event_messages_sender',
+      'entity' => 'CustomField',
+      'cleanup' => 'always',
+      'update' => 'unmodified',
+      'params' => [
+        'version' => 4,
+        'values' => [
+          'custom_group_id.name' => 'event_messages_settings',
+          'name' => 'event_messages_sender',
+          'label' => E::ts('Sender'),
+          'data_type' => 'EntityReference',
+          'html_type' => 'Autocomplete-Select',
+          'fk_entity' => 'SiteEmailAddress',
+          'filter' => 'is_active=1',
+          'fk_entity_on_delete' => 'set_null',
+          'is_searchable' => TRUE,
+          'column_name' => 'sender',
+          'in_selector' => TRUE,
+        ],
+        'match' => [
+          'custom_group_id',
+          'name',
+        ],
+      ],
+    ]
+    : [
+      'name' => 'CustomGroup_event_messages_settings_CustomField_event_messages_sender',
```

Comment:
On update from CiviCRM < 6.0.0 to >= 6.0.0 this field will be replaced by the one above. Does that work without any update code, i.e. the configured senders won't get lost?

Reply:
Good question, I haven't tested that. The same field should however be an update instead of a delete/create … Seems we have to verify that before merging.